pkger/service.go

```diff
@@ -654,75 +659,74 @@ func (s *Service) DryRun(ctx context.Context, orgID, userID influxdb.ID, pkg *Pk
 	if len(opt.EnvRefs) > 0 {
 		err := pkg.applyEnvRefs(opt.EnvRefs)
 		if err != nil && !IsParseErr(err) {
-			return Summary{}, Diff{}, internalErr(err)
+			return Summary{}, Diff{}, nil, internalErr(err)
 		}
 		parseErr = err
 	}
 
+	state := newStateCoordinator(pkg)
+
 	if opt.StackID > 0 {
-		if err := s.addStackPkgState(ctx, opt.StackID, pkg); err != nil {
-			return Summary{}, Diff{}, internalErr(err)
+		if err := s.addStackState(ctx, opt.StackID, pkg, state); err != nil {
+			return Summary{}, Diff{}, nil, internalErr(err)
 		}
 	}
 
 	if err := s.dryRunSecrets(ctx, orgID, pkg); err != nil {
-		return Summary{}, Diff{}, err
+		return Summary{}, Diff{}, nil, err
 	}
 
-	diff := Diff{
-		Buckets:    s.dryRunBuckets(ctx, orgID, pkg),
-		Checks:     s.dryRunChecks(ctx, orgID, pkg),
-		Dashboards: s.dryRunDashboards(pkg),
-		Labels:     s.dryRunLabels(ctx, orgID, pkg),
-		Tasks:      s.dryRunTasks(pkg),
-		Telegrafs:  s.dryRunTelegraf(pkg),
-		Variables:  s.dryRunVariables(ctx, orgID, pkg),
-	}
+	s.dryRunBuckets(ctx, orgID, state.mBuckets)
+	s.dryRunLabels(ctx, orgID, state.mLabels)
+
+	var diff Diff
+	diff.Checks = s.dryRunChecks(ctx, orgID, pkg)
+	diff.Dashboards = s.dryRunDashboards(pkg)
+	diff.Tasks = s.dryRunTasks(pkg)
+	diff.Telegrafs = s.dryRunTelegraf(pkg)
+	diff.Variables = s.dryRunVariables(ctx, orgID, pkg)
 
 	diffEndpoints, err := s.dryRunNotificationEndpoints(ctx, orgID, pkg)
 	if err != nil {
-		return Summary{}, Diff{}, err
+		return Summary{}, Diff{}, nil, err
 	}
 	diff.NotificationEndpoints = diffEndpoints
 
 	diffRules, err := s.dryRunNotificationRules(ctx, orgID, pkg)
 	if err != nil {
-		return Summary{}, Diff{}, err
+		return Summary{}, Diff{}, nil, err
 	}
 	diff.NotificationRules = diffRules
 
-	diffLabelMappings, err := s.dryRunLabelMappings(ctx, pkg)
+	stateLabelMappings, err := s.dryRunLabelMappingsV2(ctx, state)
```

Comment:
we have the original label mapping implementation, seen right below, and this new one. They are separate for the time being. When all resources are using the new model, the original will get nuked and v2 renamed.